volatility3/framework/plugins/linux/kmsg.py

```diff
@@ -135,8 +135,14 @@ def symtab_checks(cls, vmlinux: interfaces.context.ModuleInterface) -> bool:
             bool: True if the kernel being analyzed fulfill the class requirements.
         """
 
-    def get_string(self, addr: int, length: int) -> str:
-        txt = self._context.layers[self.layer_name].read(addr, length)  # type: ignore
+    def get_string(self, addr: int, length: int) -> Union[str, None]:
+        layer = self._context.layers[self.layer_name]
+        if not layer.is_valid(addr, length):
+            vollog.error("Failed to read log record at address 0x%x", addr)
```

Comment:
This throws a pretty big error, is it not a warning, or even just an info (not displayed by default)?  How big an issue is the string being invalid?

Reply:
Hehe, not that big, which is why it wasn't considered initially. But your other comment made me think you might want this to be visible. This change was only to report the error, the code already handled this because an empty string results in an empty list when `"".splitlines()` is called.
So, a `warning` is fine?

Reply:
Ok, I just can't tell how often it might flash up for the user, but if you think it's worth their seeing during a normal run then warning is fine...

Reply:
done